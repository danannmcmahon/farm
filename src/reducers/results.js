const ini = {
    "results":[]
};

const resultsReducer = (state = ini, action = {}) => {
    console.log(action);
    switch(action.type){
        case 'UPRES':
            state= {
                ...state,
                results : action.payload
            }
            return state;
        default:
            return state;
    }
}

export default resultsReducer;