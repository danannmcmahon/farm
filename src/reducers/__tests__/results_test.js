import resultsReducer from '../results'
import { ini } from '../results'

describe('results reducer', () => {
  it('should return the initial state', () => {
    expect(resultsReducer(undefined, {})).toMatchSnapshot()
  })

  it('should (update results) handle UPRES', () => {
    expect(
      resultsReducer(ini,
      {
        type: 'UPRES',
        payload:{"Test":"Test Value"}
      })
    ).toMatchSnapshot()
  })
})