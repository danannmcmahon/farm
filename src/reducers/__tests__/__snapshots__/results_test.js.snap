// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`results reducer should (update results) handle UPRES 1`] = `
Object {
  "results": Object {
    "Test": "Test Value",
  },
}
`;

exports[`results reducer should return the initial state 1`] = `
Object {
  "results": Array [],
}
`;
